pkg/crc/api/api.go

```diff
@@ -69,12 +90,22 @@ func (api CrcApiServer) handleConnection(conn net.Conn) {
 		logging.Error("Error decoding request: ", err.Error())
 		return
 	}
-
-	if handler, ok := api.handlers[req.Command]; ok {
-		result := handler(req.Args)
-		writeStringToSocket(conn, result)
+	if req.Command == "start" || req.Command == "stop" || req.Command == "delete" {
+		// queue new request to channel
+		if !addRequestToChannel(req, api.clusterOpsRequestsChan) {
+			defer conn.Close()
+			logging.Error("Channel capacity reaced, unable to add new request")
+			writeStringToSocket(conn, "Channel capacity reaced, unable to add new request")
+			return
+		}
+		if !addSocketsToChannel(conn, api.socketsChan) {
+			defer conn.Close()
+			logging.Error("Channel capacity reaced, unable to add new request")
+			writeStringToSocket(conn, "Channel capacity reaced, unable to add new request")
+			return
+		}
 	} else {
-		writeStringToSocket(conn, fmt.Sprintf("Unknown command supplied: %s", req.Command))
+		go api.handleRequest(req, conn)
```

Comment:
Yes this has to be run async, since we need to handle the other requests as well.

> The bigger question is whether they can be run concurrently with start/stop/.. operations.

Yes, there shouldn't be any issues running the status, version, and webconsoleurl command concurrently, in case they fail we handle the error. and it shouldn't kill the daemon process.

Also to implement polling to get the status update periodically and since a GUI is involved we need to offer these to be run even when a time consuming background process is on going in the background.

Reply:
I'm still *very* uncomfortable running concurrently code which was not written for that, but I can't spot potential problems either so..